.github/workflows/yapf.yml

```diff
@@ -0,0 +1,24 @@
+name: run yapf on source code
+
+on: push
+
+jobs:
+  yapf:
+    runs-on: ubuntu-latest
+    steps:
+    # https://github.com/marketplace/actions/setup-miniconda
+    - uses: actions/checkout@v2
+    - uses: goanpeca/setup-miniconda@v1
+      with:
+         auto-update-conda: true
+         activate-environment: anaconda-client-env
+         python-version: 3.8
+         auto-activate-base: false
+    - name: Install yapf
+      shell: bash -l {0}
+      run: |
+        pip install yapf
+    - name: run yapf
+      shell: bash -l {0}
+      run: |
+        yapf -r --diff .
```

Comment:
Does it exit with a non zero status if there is some diff?

Reply:
Yes I believe it does, the `--diff` option does that and this is what the manual suggests in ci

```
(base) (icubam)romain@plume:~/inria/icubam$ cat foo.py
def foo():
        print("hi")
(base) (icubam)romain@plume:~/inria/icubam$ yapf -r --diff . && echo Previous command terminated cleanly
--- ./foo.py	(original)
+++ ./foo.py	(reformatted)
@@ -1,2 +1,2 @@
 def foo():
-        print("hi")
+  print("hi")

```